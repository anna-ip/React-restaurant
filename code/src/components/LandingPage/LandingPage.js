import React from 'react'
import { Link } from 'react-router-dom'
import Reviews from './ReviewContainer/Reviews'
import styles from './LandingPage.module.scss'
import mainImage from '../../images/pexels-emre-can-2079438.jpg'

const LandingPage = () => {
  return (
    <div className={styles.landingpageContainer}>
      <div className={styles.linkContainer}>
        <Link to="/menu" className={styles.link}>
          Take me to the
        </Link>
        <Link to="/menu" className={styles.menu}>
          Menu
        </Link>
      </div>
      <img src={mainImage} alt="restaurant" width="100%" height="auto" />
      <Reviews />
    </div>
  )
}

export default LandingPage
