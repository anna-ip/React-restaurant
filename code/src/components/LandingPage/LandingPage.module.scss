@import '../../styles/_variables.scss';
@import '../../styles/mixins';

.landingpageContainer {
  @include flexCenter;
}
.linkContainer {
  @include flexCenter;
  justify-content: center;
  height: 5rem;
}
.img {
  width: 90%;
  height: auto;
  border: $image-border;
}
