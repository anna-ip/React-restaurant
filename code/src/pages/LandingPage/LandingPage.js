import React from 'react'
import Reviews from '../../components/ReviewComponents/REviewComponents/Reviews/Reviews'
import styles from './LandingPage.module.scss'
import mainImage from '../../images/pexels-emre-can-2079438.jpg'

const LandingPage = () => {
  return (
    <div className={styles.landingPageContainer}>
      <img src={mainImage} alt="restaurant" className={styles.img} />
      <Reviews />
    </div>
  )
}

export default LandingPage
