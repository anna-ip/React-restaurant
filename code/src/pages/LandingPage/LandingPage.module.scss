@import '../../styles/_variables.scss';
@import '../../styles/mixins';

.landingPageContainer {
  @include flexCenter;
}

.img {
  width: 90%;
  height: auto;
  @include imageBorder;
}
