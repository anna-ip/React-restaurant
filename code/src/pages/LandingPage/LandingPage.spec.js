import React from 'react'
import LandingPage from './LandingPage'
import { shallow } from 'enzyme'
import Image from '../../images/pexels-emre-can-2079438.jpg'

describe('<LandingPage/>', () => {
  test('Renders correctly using enzyme', () => {
    const tree = shallow(<LandingPage />)
    expect(tree).toMatchSnapshot()
  })

  // Test Review component?
  // Test there is an image?
  console.log('Image:', Image)
  test.skip('renders image', () => {
    const wrapper = shallow(<LandingPage />)
    const image = wrapper.find('.img')

    expect(image.src).toEqual(Image)
  })
})
