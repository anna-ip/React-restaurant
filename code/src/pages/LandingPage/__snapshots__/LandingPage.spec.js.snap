// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> Renders correctly using enzyme 1`] = `
<div
  className="landingPageContainer"
>
  <img
    alt="restaurant"
    className="img"
    src="pexels-emre-can-2079438.jpg"
  />
  <Reviews />
</div>
`;
